bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ProfileContextImpl.java

```diff
@@ -12,26 +12,42 @@
  */
 package org.openhab.core.thing.internal;
 
+import java.util.List;
 import java.util.concurrent.ScheduledExecutorService;
 
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.openhab.core.common.ThreadPoolManager;
 import org.openhab.core.config.core.Configuration;
 import org.openhab.core.thing.profiles.ProfileContext;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
 
 /**
  * {@link ProfileContext} implementation.
  *
  * @author Simon Kaufmann - Initial contribution
+ * @author Jan N. Klug - Add accepted type methods
  */
 @NonNullByDefault
 public class ProfileContextImpl implements ProfileContext {
 
     private static final String THREAD_POOL_NAME = "profiles";
     private final Configuration configuration;
 
+    private final List<Class<? extends State>> acceptedDataTypes;
+    private final List<Class<? extends Command>> acceptedCommandTypes;
+
     public ProfileContextImpl(Configuration configuration) {
         this.configuration = configuration;
+        this.acceptedDataTypes = List.of();
+        this.acceptedCommandTypes = List.of();
+    }
```

Comment:
Should we reuse the second ctor here?

```suggestion
    public ProfileContextImpl(Configuration configuration) {
        this(configuration, List.of(), List.of());
    }
```

Reply:
Done.